docs/guide/profiling-test-performance.md

```diff
@@ -0,0 +1,81 @@
+# Profiling Test Performance
+
+When you run Vitest it reports multiple time metrics of your tests:
+
+> ```bash
+> RUN  v2.1.1 /x/vitest/examples/profiling
+>
+> ✓ test/prime-number.test.ts (1) 4517ms
+>   ✓ generate prime number 4517ms
+>
+> Test Files  1 passed (1)
+>      Tests  1 passed (1)
+>   Start at  09:32:53
+>   Duration  4.80s (transform 44ms, setup 0ms, collect 35ms, tests 4.52s, environment 0ms, prepare 81ms)
+>   # Time metrics ^^
+> ```
+
+- Transform: How much time was spent transforming the files.
+- Setup: Time spent for running the [`globalSetup`](/config/#globalsetup) files.
+- Collect: Time spent for colleting all tests in the test files. This includes the time it took to import all file dependencies.
+- Tests: Time spent for actually running the test cases.
+- Environment: Time spent for setting up the test environment, for example JSDOM.
+- Prepare: ???
```

Comment:
Is the setup one correct, or is that for `setupFiles` instead? How about prepare time?

Reply:
"prepare" is the time it takes for Vitest to actually start working with the user code (the test or a setup file). Basically, it's a difference between the time when the worker was initiated and the time when we start importing the setup file or a test file (if there is no setup file). During that period, Vitest sets up common env, starts coverage inside worker, imports the test runner, snapshot environment, and the vitest entry point 

It's an internal metric that we need to look after, not the user. It's just there to catch regressions we might've introduced

Reply:
I'll add here `- Prepare: Time Vitest uses to prepare the test runner.`. Thanks for help!